### PR TITLE
Fix regression in system UI colors

### PR DESCRIPTION
**Pull request by Piinks**
Fixes https://github.com/flutter/flutter/issues/88431

This fixes a regression introduced in https://github.com/flutter/engine/pull/27018. The refactored style portion made color changes dependent on brightness not being null.

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [Flutter Style Guide] and the [C++, Objective-C, Java style guides].
- [x] I listed at least one issue that this PR fixes in the description above.
- [x] I added new tests to check the change I am making or feature I am adding, or Hixie said the PR is test-exempt. See [testing the engine] for instructions on
writing and running engine tests.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I signed the [CLA].
- [x] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[C++, Objective-C, Java style guides]: https://github.com/flutter/engine/blob/master/CONTRIBUTING.md#style
[testing the engine]: https://github.com/flutter/flutter/wiki/Testing-the-engine
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat

### CODE REVIEW

File: shell/platform/android/io/flutter/plugin/platform/PlatformPlugin.java [hunk 1]
```diff
@@ -374,15 +374,17 @@ private void setSystemChromeSystemUIOverlayStyle(
...
-      switch (systemChromeStyle.statusBarIconBrightness) {
-        case DARK:
-          // View.SYSTEM_UI_FLAG_LIGHT_STATUS_BAR
-          flags |= 0x2000;
-          break;
-        case LIGHT:
-          flags &= ~0x2000;
-          break;
+    if (Build.VERSION.SDK_INT >= 23) {
+      if (systemChromeStyle.statusBarIconBrightness != null) {
```
Comment (blasten) [inline shell/platform/android/io/flutter/plugin/platform/PlatformPlugin.java:378]: is this mostly a style change? like break a single `if` into two?

Comment (blasten) [reply shell/platform/android/io/flutter/plugin/platform/PlatformPlugin.java:378]: i see nvm

File: shell/platform/android/io/flutter/plugin/platform/PlatformPlugin.java [hunk 2]
```diff
@@ -374,15 +374,17 @@ private void setSystemChromeSystemUIOverlayStyle(
...
+        switch (systemChromeStyle.statusBarIconBrightness) {
+          case DARK:
+            // View.SYSTEM_UI_FLAG_LIGHT_STATUS_BAR
```
Comment (blasten) [inline shell/platform/android/io/flutter/plugin/platform/PlatformPlugin.java:381]: This value was [deprecated](https://developer.android.com/reference/android/view/View#SYSTEM_UI_FLAG_LIGHT_STATUS_BAR) in API 30.   

Do we know if this still works in Build.VERSION.SDK_INT >= 30 ?

File: shell/platform/android/io/flutter/plugin/platform/PlatformPlugin.java [hunk 3]
```diff
@@ -403,16 +405,17 @@ private void setSystemChromeSystemUIOverlayStyle(
...
-          flags |= 0x10;
-          break;
-        case LIGHT:
-          flags &= ~0x10;
-          break;
+    if (Build.VERSION.SDK_INT >= 26) {
+      if (systemChromeStyle.systemNavigationBarIconBrightness != null) {
+        switch (systemChromeStyle.systemNavigationBarIconBrightness) {
+          case DARK:
+            // View.SYSTEM_UI_FLAG_LIGHT_NAVIGATION_BAR
```
Comment (blasten) [inline shell/platform/android/io/flutter/plugin/platform/PlatformPlugin.java:412]: This one is also [deprecated](https://developer.android.com/reference/android/view/View#SYSTEM_UI_FLAG_LIGHT_NAVIGATION_BAR).

File: shell/platform/android/io/flutter/plugin/platform/PlatformPlugin.java (hunk 2 above)
Comment (Piinks) [reply shell/platform/android/io/flutter/plugin/platform/PlatformPlugin.java:381]: Yeah, the migration should probably be in a separate change though so all of this file can be migrated.

**Review by blasten — APPROVED** [review]
lgtm

**Pull request merged**